packages/template-retail-react-app/app/utils/responsive-image.js

```diff
@@ -1,102 +1,311 @@
-            src: getSrcWithoutOptionalParams(src)
+function obtainImageLinkMedia(breakpointIndex) {
+    const toMediaValue = (bp, type) => {
+        const val = themeBreakpoints[bp]
+        if (emValue.test(val)) {
+            // em value
+            const parsed = parseFloat(val)
+            return {[type]: type === 'max' ? `${parsed - 0.01}em` : `${parsed}em`}
         }
+
+        const parsed = parseInt(val, 10)
+        return {[type]: type === 'max' ? `${parsed - 1}px` : `${parsed}px`}
     }
 
-    themeBreakpoints = breakpoints
-    breakpointLabels = getBreakpointLabels(themeBreakpoints)
+    const nextBp = breakpointLabels.at(breakpointIndex + 1)
+    const currentBp = breakpointLabels.at(breakpointIndex)
+    if (breakpointIndex === 0) {
+        // first
+        return toMediaValue(nextBp, 'max')
+    } else if (breakpointIndex === breakpointLabels.length - 1) {
+        // last
+        return toMediaValue(currentBp, 'min')
+    }
+    return {...toMediaValue(currentBp, 'min'), ...toMediaValue(nextBp, 'max')}
+}
 
-    // Order of these attributes matter! If src is not last, Safari will refetch images
-    // multiple times (once when it processes src and again when it processes sizes / srcSet)
-    // See https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2223
-    return {
-        sizes: mapWidthsToSizes(widths),
-        srcSet: mapWidthsToSrcSet(widths, src),
-        src: getSrcWithoutOptionalParams(src)
+/**
+ * Ensures the mapping of various supported `densities` input types to
+ * a unified internal output/representation that allows influencing an
+ * image's effectively requested size as well as its display width.
+ * The resulting data shape contains an array of numeric tuples, where
+ * each tuple entry represents conversion factors to apply for certain
+ * targeted theme breakpoints.
+ *
+ * At the end, the `densities` work in conjunction with the `widths`
+ * property passed to the {@link getResponsivePictureAttributes} method
+ * as well. While the `widths` define the visual space/size an image should
+ * consume on the screen when a certain breakpoint matches, the `densities`
+ * allow fine-granular control each image's properties.
+ *
+ * Per breakpoint, it's possible to define multiple density factor tuples.
+ * In this way, several images can be defined per breakpoint, optimized to
+ * be loaded by devices with different matching resolutions.
+ *
+ * The computation per theme breakpoint works as follows. Example for breakpoint
+ * `lg` (80em):
+ * 1. The related `widths` entry (e.g. `50vm`) is used to compute a `px` value.
+ *    Example: 80 * 16 * 50vm = 1280 / 2 = 640 (`px`)
+ * 2. The related `densities` factor entry (e.g. `[0.75, 1.5]`) is used to compute the final image dimensions.
+ *    Example: display width = 640 * 0.75 = 480; DIS size = 640 * 2 = 960
+ * 3. The resulting image definition then looks like `http://example.com/image.jpg?sw=960&q=60 480w`
+ * @example Array of numeric values
```

Comment:
The jsdoc comments is longer than the implementation of the function itself. Can we shorten these examples, and the explanation into 3 paragraph max. This seems a bit excessive

Reply:
I'm sorry to say that, but this approach/assumption cannot be more wrong. The intent needs to be to have way more of such detailed descriptions/comments in the codebase. We're meanwhile not only doing this for human's sake, but more importantly to also give AI tools all the context they need to operate effectively and correctly. As we're still using ancient JS code in this package/workspace, we need all this context about supported types, etc. to be part of the JSDoc.